### PR TITLE
Split the Cmd and Ctrl mod keys from keyboard shortcuts

### LINKED ISSUE #9369: Distinguish the meta keys (Cmd/Ctrl) in keyboard shortcuts depending on the OS
### Description
The issue is about distinguishing the meta keys (Cmd/Ctrl) depending on the OS. In v11.1 and below pressing the meta keys like Command or Control are treated the same. For example, the keyboard shortcut such as `Cmd + Arrow left` can be triggered on the macOS as well as `Ctrl + Arrow left`.

Within the issue, the behavior will be changed. On macOS only the Command meta key should be used to trigger related shortcuts and on other OS like Windows, Linux only the Control meta key should be tracked.

The issue is part of the https://github.com/handsontable/handsontable/issues/9361.

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR splits the **Command ⌘** and **Control** modifier keys to separate independent keys. From now on, the table would trigger its keyboard shortcuts action differently on macOS and non-macOS systems. 

For example for `Cmd ⌘/Ctrl + A`&sup1;:

&nbsp; | macOS | non-macOS
-- | -- | --
v11.1 | both works Cmd ⌘/Ctrl + A | both works Cmd ⌘/Ctrl + A
v12.0 (this PR) | Cmd ⌘ + A | Ctrl + A

&sup1;The PR's change affects all keyboard shortcuts that use the Cmd ⌘ or Ctrl modifier keys.

To avoid DRY, the Shotcut module from now on accepts a virtual shortcut key called `'control/meta'`. That shortcut encapsulates the differences in keyboard layouts between macOS and non-macOS systems. So this code:
```js
hot.getShortcutManager().getContext('grid').addShortcut({
  keys: [[isMacOS() ? 'Meta' : 'Command', 'K']],
  group: 'edit-group',
  callback() {
    console.info('action');
  }
})
```
can be simplified to:
```js
hot.getShortcutManager().getContext('grid').addShortcut({
  keys: [['Command/Meta', 'K']],
  group: 'edit-group',
  callback() {
    console.info('action');
  }
})
```

It's still possible to define keyboard shortcuts using the meta keys such as `Command` and `Meta` independently. Removing or adding those combinations to the shortcut keys collection does not interfere with shortcuts that use the virtual key.

### How has this been tested?

I tested the changes (using a small slice of keyboard shortcuts) locally on macOS, Win 7, and Ubuntu 21. I didn't find any issues. Moreover, I added new unit and E2E tests.

### Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Related issue(s):
1. fixes #9369

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/shortcuts/manager.js [hunk 1]
```diff
@@ -83,29 +82,35 @@ export const createShortcutManager = ({ ownerWindow, beforeKeyDown, afterKeyDown
    */
   const keyRecorder = useRecorder(ownerWindow, beforeKeyDown, afterKeyDown, (event, keys) => {
     const activeContext = getContext(getActiveContextName());
+    let isExecutionCancelled = false;
 
     if (activeContext.hasShortcut(keys)) {
       // Processing just actions being in stack at the moment of shortcut pressing (without respecting additions/removals performed dynamically).
-      const shortcuts = activeContext.getShortcuts(keys).slice();
+      const shortcuts = activeContext.getShortcuts(keys);
```
Comment (wszymanski) [inline handsontable/src/shortcuts/manager.js:89]: Why do you remove it? I'm surprised the tests pass. The above comments show a purpose for `slice`.

Comment (budnix) [reply handsontable/src/shortcuts/manager.js:89]: The [`getShortcuts`](https://github.com/handsontable/handsontable/blob/feature/issue-9369/handsontable/src/shortcuts/context.js#L178) method returns always a new copy of an array. The `slice` call is redundant here, I guess.

**Review by wszymanski — APPROVED** [review]
:tada LGFM. Please remember to add documentation about the feature.

**Comment by budnix** [comment]
The PR is accepted, so I guess @wszymanski agrees with the name of the virtual shortcut `Command/Meta`. @kirszenbaum, Can I ask you to mention this in our docs? I think that the PR's description already describes everything that you need?

**Pull request merged**